Extension/src/LanguageServer/ui_old.ts

```diff
@@ -0,0 +1,442 @@
+            + (twoStatus ? " | " : "")
+            + (val ? this.workspaceParsingStatus : "");
+    }
+
+    private setIsParsingWorkspacePausable(val: boolean): void {
+        if (val) {
+            this.browseEngineStatusBarItem.command = "C_Cpp.ShowParsingCommands";
+        } else {
+            this.browseEngineStatusBarItem.command = undefined;
+        }
+    }
+
+    private setIsParsingWorkspacePaused(val: boolean): void {
+        this.isParsingWorkspacePaused = val;
+    }
+
+    private setIsCodeAnalysisPaused(val: boolean): void {
+        if (!this.isRunningCodeAnalysis) {
+            return;
+        }
+        this.isCodeAnalysisPaused = val;
+        const twoStatus: boolean = val && this.isUpdatingIntelliSense;
+        this.intelliSenseStatusBarItem.tooltip = (this.isUpdatingIntelliSense ? this.updatingIntelliSenseTooltip : "")
+            + (twoStatus ? " | " : "")
+            + (val ? this.codeAnalysisPausedTooltip : this.runningCodeAnalysisTooltip);
+    }
+
+    private setIsParsingFiles(val: boolean): void {
+        this.isParsingFiles = val;
+        const showIcon: boolean = val || this.isParsingWorkspace;
+        const twoStatus: boolean = val && this.isParsingWorkspace;
+        this.ShowDBIcon = showIcon;
+        this.browseEngineStatusBarItem.text = showIcon ? "$(database)" : "";
+        this.browseEngineStatusBarItem.tooltip = (val ? this.parsingFilesTooltip : "")
+            + (twoStatus ? " | " : "")
+            + (this.isParsingWorkspace ? this.workspaceParsingStatus : "");
+    }
+
+    private setIsUpdatingIntelliSense(val: boolean): void {
+        this.isUpdatingIntelliSense = val;
+        const showIcon: boolean = val || this.isRunningCodeAnalysis;
+        const twoStatus: boolean = val && this.isRunningCodeAnalysis;
+        this.ShowFlameIcon = showIcon;
+        this.intelliSenseStatusBarItem.text = showIcon ? "$(flame)" : "";
+        this.intelliSenseStatusBarItem.tooltip = (val ? this.updatingIntelliSenseTooltip : "")
+            + (twoStatus ? " | " : "")
+            + (this.isRunningCodeAnalysis ? this.runningCodeAnalysisTooltip : "");
+    }
+
+    private setIsRunningCodeAnalysis(val: boolean): void {
+        if (this.isRunningCodeAnalysis && !val) {
+            this.codeAnalysisTotal = 0;
+            this.codeAnalysisProcessed = 0;
+        }
+        this.isRunningCodeAnalysis = val;
+        const showIcon: boolean = val || this.isUpdatingIntelliSense;
+        const twoStatus: boolean = val && this.isUpdatingIntelliSense;
+        this.ShowFlameIcon = showIcon;
+        this.intelliSenseStatusBarItem.text = showIcon ? "$(flame)" : "";
+        this.intelliSenseStatusBarItem.tooltip = (this.isUpdatingIntelliSense ? this.updatingIntelliSenseTooltip : "")
+            + (twoStatus ? " | " : "")
+            + (val ? this.runningCodeAnalysisTooltip : "");
+        this.intelliSenseStatusBarItem.command = val ? "C_Cpp.ShowCodeAnalysisCommands" : undefined;
```

Comment:
Is this supposed to be changed? It looks like this command got replaced by the idle/active ones. However, as mentioned elsewhere, I'm concerned about having the 2 different lists of commands, because the "idle" commands are still valid to run when non-idle, i.e. run code analysis on active file, switch file, run code analysis on active file (the new one, in parallel).